src/api/c/random_engine.cpp

```diff
@@ -0,0 +1,377 @@
+/*******************************************************
+ * Copyright (c) 2014, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#include <af/random_engine.h>
+
+#include <af/array.h>
+#include <af/data.h>
+#include <af/defines.h>
+#include <af/dim4.hpp>
+#include <backend.hpp>
+#include <err_common.hpp>
+#include <handle.hpp>
+#include <random_engine.hpp>
+#include <random.hpp>
+#include <MersenneTwister.hpp>
+#include <types.hpp>
+
+using detail::uint;
+using detail::cfloat;
+using detail::cdouble;
+using detail::uchar;
```

Comment:
According to the windows build log, you need to add `using detail::ushort`.
Most files just use `using namespace detail;`